project_1/include/server.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "packet.hpp"
+
+#include <cstdint>
+#include <vector>
+#include <queue>
+
+// Type for server receiver, contains only ID
+struct ServerBase {
```

Comment:
A class definition should usually start with a public: section, followed by protected:, then private:. Omit sections that would be empty.
https://google.github.io/styleguide/cppguide.html#Declaration_Order